portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,36 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+const addComments = async () => {
+  const commentsContainer = document.getElementById('comments-container');
+  try {
+    const response = await fetch('/data');
+    if (!response.ok) {
+      throw new Error(
+          'Error occurred while fetching comment data.', response.status,
+          response.statusText);
+    }
+    const rawComments = await response.json();
+    const commentTemplate = document.getElementById('comment-template');
+    for (let i = 0; i < rawComments.length; i++) {
```

Comment:
Use forEach instead of a for-loop here and I'd break apart the logic in to its own function and document the purpose, to turn raw comments from the Java servlet into HTML comments.